### ISSUE TITLE
Possibility for better exceptions

### ISSUE THREAD

**Issue by stewartgordonsitka** [egonl/SharpDocx#13]
Hi egonl,

Thanks for the library and the recent to the nuget package for v2, I really appreciate the work that you've put into this.

Currently we are catching the SharpDocxCompilationException and returning some helpful messages to users, but we noticed there are some exceptions thrown in the CodeBlockBuilder that are a generic System.Exception. 

While parsing the message doesn't cause too much trouble, it could be helpful to have the SharpDocxCompilationException thrown here or perhaps a new type of SharpDocxCodeBlockBuilderException.

Would you find it helpful if I submitted a pull request for something like this in the future?

Thank you

**Comment by egonl** [comment]
Good point. I think CodeBlockBuilder and DocumentCompiler should throw a SharpDocxParseException when there's an error parsing the document. What do you think? 

Oh and yes, pull requests would be much appreciated.

**Comment by MAP-Akamae** [comment]
Hi, 
      
      It would be great if the errors were returned within the message property which is a default property of every exception instead of the custom property Errors
      
   I know that the entity framework team does something similar to you, but I think this makes it more difficult than it helps to debug problems
